### PR TITLE
Import CLI

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: .github/workflows/publish-cli.yml [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        run: echo "LIBCLANG_PATH=$((gcm clang).source -replace "clang.exe")" >> $env:GITHUB_ENV
+        if: matrix.os == 'windows-latest'
+
+      # build the CLI
+      - name: Build
+        uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --profile production
+
```
Comment (Thoralf-M) [inline .github/workflows/publish-cli.yml:108]: This needs to run in the cli dir

File: .github/workflows/publish-cli.yml [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+          KEYCHAIN_PASSWORD: ${{ secrets.KEYCHAIN_PASSWORD }}
+          MAC_CERT_BASE64: ${{ secrets.MAC_CERT_BASE64 }}
+          MAC_CERT_PASSWORD: ${{ secrets.MAC_CERT_PASSWORD }}
+        if: matrix.os == 'macos-latest'
+
+      - name: Sign and notarize Wallet CLI binary (macOS)
+        run: |
+          gon gon-config.json
+          unzip wallet.zip
+          mv -f wallet target/production/wallet
```
Comment (Thoralf-M) [inline .github/workflows/publish-cli.yml:131]: This will be wrong if not in the cli dir

File: .github/workflows/publish-cli.yml [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        if: matrix.os == 'macos-latest'
+
+      - name: Delete keychain (macOS)
+        run: security delete-keychain signing.keychain
+        # Run even if previous steps fail
+        if: ${{ matrix.os == 'macos-latest' && always() }}
+
+      # Computes SHA-256 checksum
+      - name: SHA-256 checksum
+        run: shasum -a 256 "./target/production/wallet${{ matrix.ext }}"
```
Comment (Thoralf-M) [inline .github/workflows/publish-cli.yml:144]: This will be wrong if not in the cli dir

File: .github/workflows/publish-cli.yml [hunk 4]
```diff
@@ -0,0 +1,154 @@
+name: Publish
+
+on:
+  push:
+    branches:
+      - main
```
Comment (Thoralf-M) [inline .github/workflows/publish-cli.yml]:
```suggestion
      - develop
      - production
```

File: .github/workflows/publish-cli.yml [hunk 5]
```diff
@@ -0,0 +1,154 @@
+name: Publish
```
Comment (Thoralf-M) [inline .github/workflows/publish-cli.yml]:
```suggestion
name: Publish cli wallet
```

File: cli/Cargo.toml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+[[bin]]
+name = "wallet"
+path = "src/main.rs"
+
+[dependencies]
+clap = { version = "3.2.23", default-features = false, features = [ "derive", "std" ] }
+dialoguer = { version = "0.10.3", default-features = false, features = [ "password" ] }
+fern-logger = { version = "0.5.0", default-features = false }
+iota-wallet = { git = "https://github.com/iotaledger/wallet.rs", rev = "9ebfa3355af46be72ffbac62a465be122b8c325e", default-features = false, features = [ "storage", "stronghold", "participation" ] }
```
Comment (Thoralf-M) [inline cli/Cargo.toml]: Should this be changed to path = "../wallet" now?

File: cli/README.md [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+export WALLET_DATABASE_PATH=/path/to/database # or add it to your .bashrc, .zshrc
+./wallet [COMMAND] [OPTIONS]
+```
+
+## Contributing
+
+To run the CLI from source, install Rust (usually through [Rustup](https://rustup.rs/)) and run the following commands:
+
+```
+git clone https://github.com/iotaledger/cli-wallet
```
Comment (Thoralf-M) [inline cli/README.md]:
```suggestion
git clone https://github.com/iotaledger/wallet.rs
cd cli
```

File: cli/documentation/docs/00_welcome.md [hunk 8]
```diff
@@ -0,0 +1,16 @@
+# Welcome
+
+![cli-wallet](/img/cli-wallet.gif)
+
+The [cli-wallet](https://github.com/iotaledger/cli-wallet) is a stateful Command Line Interface wrapper around
```
Comment (Thoralf-M) [inline cli/documentation/docs/00_welcome.md]:
```suggestion
The [cli-wallet](https://github.com/iotaledger/wallet.rs/tree/develop/cli) is a stateful Command Line Interface wrapper around
```

File: cli/documentation/docs/01_installation.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
+# Installation
+
+## From release
+
+### 1. Download
+
+Go to https://github.com/iotaledger/cli-wallet/releases and download the latest release binary for your platform.
```
Comment (Thoralf-M) [inline cli/documentation/docs/01_installation.md]: For now probably still makes sense, but for the next release not anymore, should we already change?

File: cli/documentation/docs/01_installation.md [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+### 1. Install Rust
+
+https://www.rust-lang.org/tools/install
+
+### 2. Compile
+
+```sh
+git clone https://github.com/iotaledger/cli-wallet -b develop
+cd cli-wallet
```
Comment (Thoralf-M) [inline cli/documentation/docs/01_installation.md]:
```suggestion
git clone https://github.com/iotaledger/wallet.rs -b develop
cd wallet.rs/cli
```

File: .github/workflows/build-and-test.yml [hunk 11]
```diff
@@ -181,8 +181,8 @@ jobs:
         # https://github.com/iotaledger/wallet.rs/pull/467/checks?check_run_id=2214108362#step:11:41
```
Comment (Thoralf-M) [inline .github/workflows/build-and-test.yml:181]: Can't comment, but maybe change line 99 so it also only tests wallet?

Comment (Thoralf-M) [reply .github/workflows/build-and-test.yml:181]: Also do that in https://github.com/iotaledger/wallet.rs/blob/develop/.github/workflows/private-tangle-tests.yml

**Review by Thoralf-M — COMMENTED** [review]
https://github.com/iotaledger/wallet.rs/blob/develop/coverage.sh probably also needs to be changed

File: cli/documentation/docs/01_installation.md (hunk 9 above)
Comment (thibault-martinez) [reply cli/documentation/docs/01_installation.md]: Yeah, when it's merged we can do a new release, it's been a long time anyway

**Review by Thoralf-M — DISMISSED** [review]

**Review by Thoralf-M — DISMISSED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**